File: world-clickers/convert.js
const fs = require('fs');

const cssFile = 'style.css';
const cssBak = 'style.css.bak';

if (!fs.existsSync(cssBak)) {
    fs.copyFileSync(cssFile, cssBak);
}

let css = fs.readFileSync(cssFile, 'utf8');

// Ensure #app-container is fully removed
css = css.replace(/#app-container\s*\{[^}]+\}\s*/, '');
// Revert game wrapper to taking full viewport space (100vh/100vw instead of fixed 1080/1920)
css = css.replace(/width: 1920px;[\s\n]*position: relative;/, 'width: 100vw;\n    position: relative;');
css = css.replace(/height: 1080px;[\s\n]*width: 1920px;/, 'height: 100vh;\n    width: 100vw;');

// Process blocks
css = css.replace(/\{([^}]+)\}/g, (match, block) => {
    let newBlock = block.replace(/(-?\d*\.?\d+)px/g, (m, valStr) => {
        let val = parseFloat(valStr);
        if (val === 0) return '0px';
        return `calc(${val}px * var(--s))`;
    });
    return "{" + newBlock + "}";
});

// Add variable to root
if (!css.includes(':root {')) {
    const rootVars = `:root {\n    --s: 1;\n}\n\nhtml {\n    font-size: calc(16px * var(--s));\n}\n\n`;
    css = rootVars + css;
}

fs.writeFileSync(cssFile, css, 'utf8');
console.log("Successfully processed style.css with Node!");

File: world-clickers/convert.py
import re
import os
import shutil

css_file = 'style.css'

if not os.path.exists(css_file + '.bak'):
    shutil.copy2(css_file, css_file + '.bak')

with open(css_file, 'r', encoding='utf-8') as f:
    css = f.read()

# Remove the previously added #app-container block
# and revert the game-wrapper width/height to 100vw/100vh
css = re.sub(r'#app-container\s*\{[^}]+\}\s*', '', css)
css = css.replace('width: 1920px;\n    position: relative;\n    padding-top: 0;', 'width: 100vw;\n    position: relative;\n    padding-top: 0;')
css = css.replace('height: 1080px;\n    width: 1920px;', 'height: 100vh;\n    width: 100vw;')


def process_block(match):
    block_content = match.group(1)
    
    # regex matches e.g. "150px" or "-5.5px" or "0px"
    def replacer(m):
        full_val = m.group(0)
        val = m.group(1)
        # Skip 0px
        if float(val) == 0:
            return "0px"  # leave alone
        # Safe scale formulation
        return f"calc({val}px * var(--s))"
        
    new_content = re.sub(r'(-?\d*\.?\d+)px', replacer, block_content)
    return "{" + new_content + "}"

# Replace px sizes inside { ... } blocks to not break media queries
new_css = re.sub(r'\{([^}]+)\}', process_block, css)

# Make sure we declare the variable on the root
if ':root {' not in new_css:
    root_vars = ":root {\n    --s: 1;\n}\n\nhtml {\n    font-size: calc(16px * var(--s));\n}\n\n"
    new_css = root_vars + new_css

with open(css_file, 'w', encoding='utf-8') as f:
    f.write(new_css)
    
print("Successfully processed style.css!")
